pyNN/common/populations.py

```diff
@@ -477,8 +477,11 @@ def write_data(self, io, variables='all', gather=True, clear=False, annotations=
         file as metadata.
         """
         logger.debug("Population %s is writing %s to %s [gather=%s, clear=%s]" % (self.label, variables, io, gather, clear))
+        # ignoring the method parameter 'annotations', and using class attribute 'self.annotations'
+        # seems ok as this method parameter does not seem to be used within pynn currently.
+        # TODO: decide whether this method parameter needs to be retained.
```

Comment:
Yes: `annotations` can be supplied by the user when calling `write_data()` and `get_data()`, so this should definitely be `annotations` and not `self.annotations`